components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java

```diff
@@ -87,14 +218,59 @@ default void addNotificationTemplate(NotificationTemplate notificationTemplate,
     }
 
     /**
-     * Add a new notification template to the registry to the corresponding notification channel root directory.
+     * Update notification template.
      *
-     * @param displayName         Notification template display name
-     * @param notificationChannel Notification channel
-     * @param tenantDomain        Tenant domain
-     * @throws NotificationTemplateManagerException If an error occurred while adding the template to the registry
+     * @param notificationTemplate Notification template
+     *                             {@link org.wso2.carbon.identity.governance.model.NotificationTemplate}
+     * @param tenantDomain         Tenant domain
+     * @throws NotificationTemplateManagerException If an error occurred while updating the notification template
+     */
+    default void updateNotificationTemplate(NotificationTemplate notificationTemplate, String tenantDomain)
+            throws NotificationTemplateManagerException {
+
```

Comment:
Add not unsupported / not implemented(pending decision) exceptions if not used in other places in default form.

Reply:
https://github.com/wso2-extensions/identity-governance/pull/832/commits/9a877a5ad3ae04da62c8995e735b8b9ed4a8d3df